### PR TITLE
[NTX-334] snyk gh integration

### PR DESCRIPTION
**Pull request by HoseinEmrani**

### CODE REVIEW

**Review by elzbietaZ — APPROVED** [review]

File: .github/workflows/snyk.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            continue-on-error: false
+    name: "${{ matrix.name }}"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: snyk/actions/setup@master
+      - uses: actions/setup-go@v1
+        with:
+          go-version: '1.18'
```
Comment (elzbietaZ) [inline .github/workflows/snyk.yml:36]: Just noticed but probably applies to all this snyk prs, why 1.18 and not more recent one?

Comment (HoseinEmrani) [reply .github/workflows/snyk.yml:36]: As I mentioned to Diego in the chat, these were mostly copy & paste from a configuration that Brian made.

Comment (bshamblen) [reply .github/workflows/snyk.yml:36]: I used 1.18 on the original PR I opened for object-storage-service, since that was the version listed in the `go.mod` file.

File: .github/workflows/snyk.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: CI / snyk.io
+
+on:
+  push:
```
Comment (elzbietaZ) [inline .github/workflows/snyk.yml:4]: was wondering - why on push and not on pull_request?

Comment (HoseinEmrani) [reply .github/workflows/snyk.yml:4]: We need this to be happening on each commit push.
This will give you the  answer:
https://stackoverflow.com/a/73297089

Comment (elzbietaZ) [reply .github/workflows/snyk.yml:4]: ok, let me rephrase it, why do we need it on each commit push, not just when PR is opened? I think typically when developing something people have their branch and regularly push there, in setup like this snyk will be run every time, which seems like a lot of unnecessary runs really (I have the same feeling about our CIs setup)

Comment (HoseinEmrani) [reply .github/workflows/snyk.yml:4]: So why we run CIs every time we push but not when we open a PR? Let's say I opened a PR, and then there were many change reqs on it, and for addressing some, I introduced a package that had some critical issues, Snyk will not find that, same goes with the rest of CI steps. You can break the tests in future commits during development after you opened a PR, you can introduce linter issues, you can introduce code smells. That's why we run them on every commit. Doesn't seem unnecessary to me, that's the reason we have them.

**Review by hyxnat — APPROVED** [review]

File: .github/workflows/snyk.yml (hunk 2 above)
Comment (elzbietaZ) [reply .github/workflows/snyk.yml:4]: By `just when PR is opened` I meant any activity on the PR so it will cover what you are talking about...
